S7.Net/TPKT.cs

```diff
@@ -33,15 +33,25 @@ private TPKT(byte version, byte reserved1, int length, byte[] data)
         public static async Task<TPKT> ReadAsync(Stream stream, CancellationToken cancellationToken)
         {
             var buf = new byte[4];
-            int len = await stream.ReadExactAsync(buf, 0, 4, cancellationToken).ConfigureAwait(false);
+
+            Task<int> ansTask = stream.ReadExactAsync(buf, 0, 4, cancellationToken);
+            ansTask.Wait(stream.ReadTimeout);
```

Comment:
This blocks the thread vs. the previous async operation.

Reply:
ansTask.Wait(stream.ReadTimeout);  --> ansTask.Wait(cancellationToken);
maybe make it async code ...